lib/mocha.js

```diff
@@ -866,3 +872,46 @@ Mocha.prototype.run = function(fn) {
 
   return runner.run(done);
 };
+
+/**
+ * Assigns hooks to the root suite
+ * @param {MochaRootHookObject} [hooks] - Hooks to assign to root suite
```

Comment:
public or private?

Reply:
As it stands, it should be private, but please see [this comment](https://github.com/mochajs/mocha/pull/4244#discussion_r414029767)